src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable, named, optional } from 'inversify';
+import { SemVer } from 'semver';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { Product } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { traceWarning } from '../../../platform/logging';
+import { IsCodeSpace } from '../../../platform/common/types';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { EnvironmentType } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import {
+    IDataViewerDependencyService,
+    IDataViewerDependencyServiceOptions,
+    IdataViewerDependencyServiceOptionsWithInterpreter,
+    IDataViewerDependencyServiceOptionsWithKernel
+} from './types';
+import { executeSilently } from '../../../kernels/helpers';
+import { IKernel } from '../../../kernels/types';
+import { Identifiers } from '../../../platform/common/constants';
+import { IJupyterVariables } from '../../../kernels/variables/types';
+
+export const minimumSupportedPandaVersion = '0.20.0';
+export const kernelGetPandasVersion = [
+    'import pandas as _VSCODE_pandas;print(_VSCODE_pandas.__version__);del _VSCODE_pandas'
+];
+export const debuggerGetPandasVersion = [
+    'import pandas as _VSCODE_pandas',
+    '_VSCODE_pandas.__version__',
+    'del _VSCODE_pandas'
+];
+
+// TODO: Installing pandas does not work with the debugger.
+// It takes a long time, then it breaks with:
+// error: Command \"clang -Wno-unused-result -Wsign-compare -Wunreachable-code...
+export const debuggerInstallPandas = [
+    'import subprocess as _VSCODE_subprocess',
+    'import sys as _VSCODE_sys',
+    `def _VSCODE_install_pandas():
+  _VSCODE_subprocess.check_call([_VSCODE_sys.executable, "-m", "pip", "install", "pandas"])`,
+    '_VSCODE_install_pandas()',
+    'del _VSCODE_subprocess, _VSCODE_sys, _VSCODE_install_pandas'
+];
```

Comment:
I'm thinking on leaving this here for future reference.

Reply:
@DonJayamanne you can see the approach I took here. Here I used `sys.executable`. This code above does work — except that it crashes after a while and makes the debugger unusable.